src/udev/udev-rules-util.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "escape.h"
+#include "macro.h"
+#include "utf8.h"
+
+static inline int parse_value(char *str, char **ret_value, char **ret_endpos) {
+        char *value = str;
+        char *i, *j;
+        int r;
+        bool is_escaped;
+
+        /* value must be double quotated */
+        if (value[0] == '"') {
+                is_escaped = false;
+                value++;
+        } else if (value[0] == 'e') {
+                if (value[1] != '"')
+                        return -EINVAL;
+                is_escaped = true;
+                value += 2;
+        } else {
+                return -EINVAL;
+        }
+
+        /* unescape double quotation '\"' -> '"' */
+        for (i = j = value;; i++, j++) {
+                if (*i == '"')
+                        break;
+                if (*i == '\0')
+                        return -EINVAL;
+                if (i[0] == '\\' && i[1] == '"') {
+                        if (is_escaped) {
+                                *j = '\\';
+                                j++;
+                        }
```

Comment:
Hmm? Could you add a comment why we need to do that?

At least, this seems to cause a memory corruption, e.g., `KEY=e"\""`.

Reply:
(I believe) when `is_escaped`, the loop is used to find the end of the string. The assignment to `j` isn't needed though, as the string should not be changed. The if statement should keep the loop counting if a `"` is preceeded by a `\`.

I'll look into the memory corruption problem. Maybe make the two cases different loops is better.